src/Beacon.jsx

```diff
@@ -185,12 +204,38 @@ export class Beacon extends Component {
     }
   }
 
+  clampValueToRange(val, min, max) {
+    return Math.min(Math.max(min, val), max);
+  }
+
+  getArrowHorizontalPosition() {
+    const targetBounds = this.state.targetBounds;
+    const tooltipBounds = this.state.tooltipBounds;
+    const arrowBounds = this.state.arrowBounds;
+    const targetCenter = targetBounds.left + (targetBounds.width / 2);
+
+    // Account for animation scaling/moving the elements before it happens
+    let tooltipLeft = tooltipBounds.left;
+    if (this.state.tooltipAttachmentHorizontal === 'left') {
+      tooltipLeft = tooltipLeft + ANIMATION_MOTION_STEP/2;
+    } else if (this.state.tooltipAttachmentHorizontal === 'right') {
+      tooltipLeft = tooltipLeft - ANIMATION_MOTION_STEP/2;
```

Comment:
Found a problem here - this calculation doesn't account for `top center` or `bottom center` configuration because it scales its size from middle instead of a side. Could use some mathematical wisdom from @matt on how to write this correctly - anything below 50% arrow position should scale differently than above 50%.